### PR TITLE
feat: add list style for resource dashboard

### PR DESCRIPTION
**Pull request by CorrectRoadH**
# Screen
image

image
# Content:
## Function:
Two-style resource dashboard
cache the preferred resource dashboard style 

## DRY:
1. extract common code. such as resources Item Props
2. extract the dropdown item.

## Performance optimization:
1. using useMemo to reduce the calc of search resource

### CODE REVIEW

**Review by boojack — APPROVED** [review]
LGTM overall

File: web/src/hooks/useListStyle.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from "react";
+
+const useListStyle = () => {
+  const initialState = localStorage.getItem("listStyle") === "true";
+  const [listStyle, setListStyle] = useState(initialState);
+
+  useEffect(() => {
+    // storing input name
+    localStorage.setItem("listStyle", JSON.stringify(listStyle));
```
[the remark this answers is not in the document]
Comment (boojack) [reply web/src/hooks/useListStyle.ts:9]: Always show in grid style in first screen, and users can switch to table style.

[the remark this answers is not in the document]
Comment (boojack) [reply web/src/hooks/useListStyle.ts:9]: Make it in the simple way. WDYT

[the remark this answers is not in the document]
Comment (CorrectRoadH) [reply web/src/hooks/useListStyle.ts]: I agree with you

**Pull request merged**